src/components/lookupPublicKey/index.tsx

```diff
@@ -0,0 +1,57 @@
+import { VStack, Heading, Input, FormLabel, Button, Text } from '@chakra-ui/react';
+import { useState } from 'react';
+import { useLookupPublicKey, LookupPublicKeyStatus } from './useLookupPublicKey';
+
+export function LookupPublicKey() {
+  const [address, setAddress] = useState('');
+  const [publicKey, setPublicKey] = useState('');
+
+  const { lookupPublicKey, lookupStatus } = useLookupPublicKey();
+
+  const getLookupStatusMessage = (status: LookupPublicKeyStatus | null): string => {
```

Comment:
This is pretty minor and there is no standard for this... Generally my preference for simple cases like this is to use a map instead of switch-case for the sake of simplicity. Here we would be simply mapping `LookupPublicKeyStatus` to a string, which is a perfect case to use a map instead of switch-case. If the cases were not consistent amongst each other or had some custom logic in some cases I would keep the switch-case.

Reply:
Yes, absolutely. I think I choose this way out of habit, I wish someone would deprecate switch for me.